### PR TITLE
Enable keyboard navigation in the dropdown

### LINKED ISSUE #2342: Dropdown keyboard navigation selects an option
It should just hover it and space/enter should select it.

### PR DESCRIPTION
**Pull request by jkaltoft**
## Which issue does this PR close?

This PR closes #2342 

## What is the new behavior?
Enabling navigation with the arrow keys in the dropdown on desktop. Additionally, it's possible to use tap to focus the dropdown and open it with enter or space.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

**Review by jkaltoft — COMMENTED** [review]
Looks good to me.

However, you should get someone else to approve the PR since I contributed some of the initial changes 😁

Don't forget to change the PR title - also see https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Pull-request for more good stuff 🤓

File: libs/designsystem/src/lib/components/dropdown/dropdown-popover.component.spec.ts [hunk 1]
```diff
@@ -328,6 +328,26 @@ describe('DropdownComponent (popover version)', () => {
         });
       });
 
+      describe('and ArrowDown key is pressed', () => {
+        beforeEach(fakeAsync(() => {
+          spectator.dispatchKeyboardEvent(spectator.element, 'keydown', 'ArrowDown');
+          tick(openDelayInMs);
+        }));
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/dropdown/dropdown-popover.component.spec.ts:335]: Any reason to do beforeEach here when there is only one `it`-block?

File: libs/designsystem/src/lib/components/dropdown/dropdown.component.scss [hunk 2]
```diff
@@ -12,6 +12,11 @@ $min-screen-width: 375px;
     border-radius: utils.$border-radius-round;
   }
 
+  // // TODO: Improve this - an attempt to remove focus outline on iOS
+  // &:focus {
+  //   outline: none;
+  // }
+
```
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/dropdown/dropdown.component.scss:19]: Don't forget to remove this block comment 😎

File: libs/designsystem/src/lib/components/dropdown/dropdown-popover.component.spec.ts (hunk 1 above)
Comment (mark-drastrup) [reply libs/designsystem/src/lib/components/dropdown/dropdown-popover.component.spec.ts:335]: No I don't see any reason 😅 I'll refactor it.

File: libs/designsystem/src/lib/components/dropdown/dropdown.component.scss (hunk 2 above)
Comment (mark-drastrup) [reply libs/designsystem/src/lib/components/dropdown/dropdown.component.scss:19]: Ahh nice catch!

**Comment by mark-drastrup** [comment]
@RasmusKjeldgaard I have added some tests to check for the arrow key behaviour like we discussed 🙂

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Looks good, only one very minor comment

File: libs/designsystem/src/lib/components/item/item.component.scss [hunk 3]
```diff
@@ -92,6 +93,13 @@
   --min-height: #{utils.$dropdown-item-height};
 }
 
+// Intented for use with keyboard navigation in <kirby-dropdown>
+:host-context(kirby-dropdown .focused) ion-item {
+  --background: #{interaction-state.get-state-color('white', 'xxxs')};
+
+  // TODO: Try using Ionics custom properties (--background-focused etc)
```
[the remark this answers is not in the document]
Comment (mark-drastrup) [reply libs/designsystem/src/lib/components/item/item.component.scss]: Removed :)

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**